### DISCUSSION TITLE
Why does a snap throw an error when using URLSearchParams?

### DISCUSSION THREAD

**Discussion by einaralex** [MetaMask/snaps#656] [Q&A] [answered]
I'm trying to do a fetch inside of a snap and I need to format it with URLSearchParams.

```
await fetch('https://api.example.com/', {
      method: 'POST',
      body: new URLSearchParams({
        stuff: stuff
      }),
    });
```
and I recieved the error `MetaMask - RPC Error: URLSearchParams is not a constructor`

I tried importing URLSearchParams from 'url' and get the same result. 
I know I can do this manually, but would be interested in knowing what would be causing this, is this a bug?

**Comment by einaralex** [comment, marked as answer]
It's documented here: https://docs.metamask.io/guide/snaps-development-guide.html#developing-a-snap

>Snaps execute in a sandboxed environment that's running Secure EcmaScript (SES, see [below](https://docs.metamask.io/guide/snaps-development-guide.html#secure-ecmascript-ses)). There is no DOM, no Node.js builtins, and no platform-specific APIs other than MetaMask's wallet global object

**Reply by Montoya** [reply]
Sorry we didn't see this in time to give you an answer! Most of our team was at EthCC or on holiday :)
